create-convex-hull-around-points/src/main/java/com/esri/arcgismaps/sample/createconvexhullaroundpoints/MainActivity.kt

```diff
@@ -0,0 +1,226 @@
+        createButton.enable()
+        resetButton.enable()
+    }
+
+    /**
+     * Creates and draws a convex hull graphic on the map using [pointGeometry] points
+     */
+    private fun createConvexHull(pointGeometry: Geometry) {
+        // normalize the geometry for panning beyond the meridian
+        // and check to proceed if the resulting geometry is not null
+        GeometryEngine.normalizeCentralMeridian(pointGeometry)?.let { normalizedPointGeometry ->
+            // create a convex hull from the points and proceed if it's not null
+            GeometryEngine.convexHull(normalizedPointGeometry)?.let { convexHullGeometry ->
+                // the convex hull's geometry may be a point or polyline if the number of
+                // points is less than 3, set its symbol accordingly
+                convexHullGraphic.symbol = when (convexHullGeometry) {
+                    is Point -> {
+                        // set symbol to use the pointSymbol
+                        pointSymbol
+                    }
+                    is Polyline -> {
+                        // set symbol to use the lineSymbol
+                        lineSymbol
+                    }
+                    is Polygon -> {
+                        // set symbol to use the fillSymbol
+                        fillSymbol
+                    }
+                    else -> {
+                        showError("Unknown geometry for convex hull")
+                        null
+                    }
+                }
+                // update the convex hull graphics geometry
+                convexHullGraphic.geometry = convexHullGeometry
+                // disable the create button until new input points are created
+                createButton.disable()
+            } ?: showError("Error creating convex hull")
+        }
+    }
+
+
+    /**
+     * Resets the map by clearing any drawn points, graphics and disables all buttons
+     */
+    private fun resetMap() {
+        // remove all the selected points
+        inputPoints.clear()
+        // remove the geometry for the point graphic and convex hull graphics
+        pointGraphic.geometry = null
+        convexHullGraphic.geometry = null
+        // disabling the buttons
+        resetButton.disable()
+        createButton.disable()
+    }
+
+    private fun showError(message: String) {
+        Log.e(TAG, message)
+        Snackbar.make(mapView, message, Snackbar.LENGTH_SHORT).show()
+    }
+}
+
+/**
+ * Simple extension property that represents a blue color
+ */
+private val Color.Companion.blue
+    get() = fromRgba(0, 0, 255)
+
+/**
+ * Simple extension function to enable a button, if not already enabled
+ */
+private fun Button.enable() {
+    if (!isEnabled) isEnabled = true
+}
+
+/**
+ * Simple extension function to disable a button, if not already disabled
+ */
+private fun Button.disable() {
+    if (isEnabled) isEnabled = false
+}
```

Comment:
The idea was to not call the`button.isEnabled` setter if it's already set to the desired value, to avoid a re-draw and also for a little bit of cleanliness.

But looks like the setter actually checks for this 
`if (enabled == isEnabled()) return;`

Removing it for redundancy.

Reply:
Oh cool, I didn't know about the re-draw thing, and interesting to see that the setter already considers that.